### ISSUE TITLE
Add streetname pull-down

### ISSUE THREAD

**Issue by sjvudp** [streetcomplete/StreetComplete#5582]
**Use case**

When Street Complete asks for as street name, it's simply a free-text input field, whereas the OSM ID editor has a pull-down/autocomplete list of nearby street names. That feature saves a lot of typing, also preventing mis-spellings when entering the names on touch keyboard

**Proposed Solution**

Add pull-down list fore streetnames
Add auto-completion of streetnames
(still entering a different names should be possible

**Comment by mnalis** [comment]
> When Street Complete asks for as street name, it's simply a free-text input field,

Actually, it should have a hint just below that text field, saying something like `Tap street on the map to select it`. Does it say that for you? Have you tried tapping a nearby street?

**Comment by westnordost** [comment]
Looks like he overlooked it.

**Comment by sjvudp** [comment]
> Actually, it should have a hint just below that text field, saying something like `Tap street on the map to select it`. Does it say that for you? Have you tried tapping a nearby street?

It seems a "little arrow" is expected to say that to me, and that arrow does not always appear:

- When there is a street, but no number, it seems I get a numeric input with up/down arrows (screenshot omitted)
- Sometimes, the "little arrow" indicates a pull-down menu (and that works). I don't know the conditions, however

[Screenshot showing "pull-down" input](https://github.com/streetcomplete/StreetComplete/assets/6734808/ed8aad20-bc6b-444e-ba23-1609945b6220)

- Sometimes, no pull-down ("little arrow") appears (I admit I over-looked the hint to tap the street *on the map*):

[Screenshot showing "no pull-down" input](https://github.com/streetcomplete/StreetComplete/assets/6734808/d7559689-7518-47f8-9f70-112eff6986d9)

**Comment by westnordost** [comment]
Right, the UI is a bit inconsistent here.

For the street name quest, you have this pull-down, for the housenumber quest and overlay, you have to click on the map instead.

This inconsistency should ideally be removed (i.e. the pull-down should be removed and everywhere you can click on the map instead), however, I don't intend to work on this now as the whole UI is being re-made in another UI framework step by step. So, maybe after or during that, this could be done.

**Comment by sjvudp** [comment]
> the pull-down should be removed and everywhere you can click on the map instead

From a user-interface level of view I'm not sure whether that's an actual improvement.

**Comment by mnalis** [comment]
I agree that UI should probably be consistent (i.e. the same in both cases). 

But perhaps both options for selecting street could be kept for maximum usability and user-discoverability (i.e. in both quests, allow user to  **_either_** click on the street, **_or_** select nearby streetname from dropdown list)?
